### PR TITLE
Team statistics, add methods and test, update stat_tracker.  Tests and spec Harness pass.

### PR DESCRIPTION
**Pull request by denverdevelopments**
Add team statistics methods:  team_info, best_season, worst_season, average_win_percentage, most_goals_scored, fewest_goals_scored.  Add team stat methods to stat_trackers, and completed stat_tracker test and the corresponding game_manager_test.  All test and spec Harness successful.

### CODE REVIEW

File: lib/game_teams_manager.rb [hunk 1]
```diff
@@ -38,7 +38,7 @@ def score_and_shots_by_team(season_games_ids)
     hash
   end
 
-  def score_ratios_hash(season_games_ids)
+  def score_ratios_hash(season_games_ids)   ##Refactor?: 'hash' to 'accuracy'
```
Comment (ghost) [inline lib/game_teams_manager.rb:47]: definitely!

Comment (ahenzler) [inline lib/game_teams_manager.rb:47]: good idea to refactor.

File: lib/game_teams_manager.rb [hunk 2]
```diff
@@ -81,4 +81,24 @@ def worst_coach(season_games)
     hash.key(hash.values.min)
   end
 
+#PSUEDO  def best_season(team_id)
```
Comment (ahenzler) [inline lib/game_teams_manager.rb]: We can delete this

File: lib/games_manager.rb [hunk 3]
```diff
@@ -6,21 +6,21 @@ class GamesManager
   def initialize(data_path)
     @games = generate_list(data_path)
   end
-
```
Comment (ahenzler) [inline lib/games_manager.rb:9]: extra spaces.

File: lib/games_manager.rb [hunk 4]
```diff
@@ -87,36 +87,83 @@ def average_goals_by_season
...
+        summary[game.season] = [] if summary[game.season].nil?
+        if team_id == game.home_team_id && game.home_goals > game.away_goals
+          result = "W"
+        elsif team_id == game.away_team_id && game.home_goals < game.away_goals
+          result = "W"
+        elsif game.home_goals == game.away_goals
+          result = "T"
+        else
+          result = "L"
+        end
```
Comment (ghost) [inline lib/games_manager.rb:103]: Looks good. We'll have to figure out how to squish this down or use some helpers.

**Review by ahenzler — APPROVED** [review]

File: lib/games_manager.rb [hunk 5]
```diff
@@ -87,36 +87,83 @@ def average_goals_by_season
...
+#  require 'pry'; binding.pry
+  end
 
+  def best_season(team_id)
+    most = 0
+    best = nil
+    get_season_results(team_id).each do |key, value|   #refactor-able? #I think we can use .valuses.max/.min
+      if value.count('W').to_f / value.size > most
+        most = value.count('W').to_f / value.size
+        best = key
```
Comment (ghost) [inline lib/games_manager.rb:117]: I think I know what's happening but it's still a little hard to read.

**Pull request merged**